### PR TITLE
WT-10233 Implement dynamic mirrored table data validation in workgen

### PR DESCRIPTION
**Pull request by tammybailey**
I added a new script, `validate_mirror_tables.py` to compare the mirrored tables generated by workgen. This script uses the existing functionality found in `wt_cmp_uri`. I modified the `wt_cmp_uri` script (and renamed it to `wt_cmp_uri.py`) so that I could import it into `validate_mirror_tables.py`, hopefully without affecting the current usage of the script.

### CODE REVIEW

**Comment by tammybailey** [comment]
@ddanderson, would you please take a look at the proposed changes I made to your `wt_cmp_uri` script when you have time? I updated `wt_cmp_dir` to reflect the filename change and tested that it works as expected, but I am not sure if you are using it elsewhere or if these changes break any future work you have planned. This is just a draft, happy to change my approach if you need `wt_cmp_uri` as originally written!

File: bench/workgen/validate_mirror_tables.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    mirror = None
+
+    if result:
+        app_metadata = dict((a.strip(), b.strip())  
+            for a, b in (element.split('=')  
+                for element in result[0].split(',')))  
+
+        if app_metadata['workgen_dynamic_table'] == 'true':
+            mirror = app_metadata['workgen_table_mirror']
+            is_base = (app_metadata['workgen_base_table'] == 'true')
```
Comment (ddanderson) [inline bench/workgen/validate_mirror_tables.py:94]: It doesn't look like `is_base` is used after assignment.

**Review by ddanderson — APPROVED** [review]
This looks good.  I made some comments to followup on.  I don't object to the `.py` rename.

File: bench/workgen/validate_mirror_tables.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+def get_mirrors(db_dir, db_files):
+
+    mirrors = []
+    db_files_remaining = set(db_files)
+
+    for uri in db_files:
+        if uri not in db_files_remaining:
+            continue
+        db_files_remaining.remove(uri)
+        mirror_uri = get_mirror_uri(db_dir, f'file:{uri}')
```
[the remark this answers is not in the document]
Comment (tammybailey) [reply bench/workgen/validate_mirror_tables.py:65]: Officially (according to our wiki) the minimum Python version "required by WiredTiger and its associated test code and utilities" is Python 3.7.x. I'm happy to revert to the `.format` syntax if we have folks running older Python versions!

File: bench/workgen/validate_mirror_tables.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    files = [f for f in os.listdir(dir) if f.endswith('.wt')]
+    return files
+
+# Given a list of database files, return a list of mirrored uris.
+def get_mirrors(db_dir, db_files):
+
+    mirrors = []
+    db_files_remaining = set(db_files)
+
+    for uri in db_files:
```
[the remark this answers is not in the document]
Comment (tammybailey) [reply bench/workgen/validate_mirror_tables.py]: Changed all `uri` mentions to `file`/`filename` (where it made sense to do so) in [a118a72](https://github.com/wiredtiger/wiredtiger/pull/8865/commits/a118a722a9d587ca8ee48f295b078a47e67bb912).

File: bench/workgen/validate_mirror_tables.py (hunk 1 above)
Comment (tammybailey) [reply bench/workgen/validate_mirror_tables.py]: I missed that one, thanks! Removed in [cb52db0](https://github.com/wiredtiger/wiredtiger/pull/8865/commits/cb52db0db127d5b522d0e065730f54c8081a9ce7).

File: tools/wt_cmp_uri.py [hunk 4]
```diff
@@ -30,9 +30,10 @@
 
 import os, sys
 from wt_tools_common import wiredtiger_open
+from wiredtiger import WT_NOTFOUND, wiredtiger_strerror
```
[the remark this answers is not in the document]
Comment (tammybailey) [reply tools/wt_cmp_uri.py:33]: Yes indeed! No `PYTHONPATH` needed if you run from the build directory, and it kindly informs you to run from the build directory if you try from somewhere else.

```
$ ~/wiredtiger/build echo $PYTHONPATH

$ ~/wiredtiger/build ../tools/wt_cmp_uri.py /Users/tammy/data/file:Xl8Kv_mirror.wt /Users/tammy/data/file:Xl8Kv.wt
64 records read
$ ~/wiredtiger/build
```

File: bench/workgen/validate_mirror_tables.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ddanderson) [reply bench/workgen/validate_mirror_tables.py]: Don't revert it.  Glad to hear we're at Python 3.7.

**Comment by ddanderson** [comment]
@tammybailey Thanks, all looks good to me!

**Pull request merged**